server/services/diskUsageService.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * This service is not per rtorrent session, which is why it does not inherit
+ * `BaseService` nor have any use of the per user API ie. `getSerivce()`
+ */
+const EventEmitter = require('events');
+const util = require('util');
+const execFile = util.promisify(require('child_process').execFile);
+const config = require('../../config');
+const diskUsageServiceEvents = require('../constants/diskUsageServiceEvents');
+
+const PLATFORMS_SUPPORTED = ['darwin', 'linux'];
+
+const filterMountPoint =
+  config.diskUsageService && config.diskUsageService.watchMountPoints
+    ? // if user has configured watchPartitions filter each line output for given
+      // array
+      mountpoint => config.diskUsageService.watchMountPoints.includes(mountpoint)
+    : () => true; // include all mounted file systems by default
+
+const diskUsage = {
+  linux: () =>
+    execFile('df --block-size=1024 --portability | tail -n+2', {
+      shell: true,
+      maxBuffer: 4096,
+    }).then(({stdout}) =>
+      stdout
+        .trim()
+        .split('\n')
+        .map(disk => disk.split(/\s+/))
+        .filter(disk => filterMountPoint(disk[5]))
+        .map(([_fs, size, used, avail, _pcent, target]) => {
+          return {
+            size: Number.parseInt(size, 10) * 1024,
+            used: Number.parseInt(used, 10) * 1024,
+            avail: Number.parseInt(avail, 10) * 1024,
+            target,
+          };
+        }),
+    ),
+  darwin: () =>
+    execFile('df -kl | tail -n+2', {
+      shell: true,
+      maxBuffer: 4096,
+    }).then(({stdout}) =>
+      stdout
+        .trim()
+        .split('\n')
+        .map(disk => disk.split(/\s+/))
+        .filter(disk => filterMountPoint(disk[8]))
+        .map(([_fs, size, used, avail, _pcent, _iused, _ifree, _piused, target]) => {
+          return {
+            size: Number.parseInt(size, 10) * 1024,
+            used: Number.parseInt(used, 10) * 1024,
+            avail: Number.parseInt(avail, 10) * 1024,
+            target,
+          };
+        }),
+    ),
+  // TODO:
+  win32: () => Promise.resolve([]),
```

Comment:
Here is the command for get disks on windows:
`wmic logicaldisk get name, freespace, size`
```FreeSpace     Name  Size
115519344640  C:    201439305728
              D:
425615523840  Y:    1998251409408
26653650944   Z:    298000031744
```

Reply:
hi @jplot! Can you see how this little snippet runs on windows?

https://gist.github.com/sktt/0b89d0c90ed91825e865cf2ce35732af

@jfurrow i guess this could be added at a later stage though..